packages/boot/test/acceptance/controller.booter.app.acceptance.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright IBM Corp. 2013,2018. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Client, createClientForHandler, TestSandbox} from '@loopback/testlab';
+import {RestServer} from '@loopback/rest';
+import {BootOptions} from '@loopback/core';
+import {ControllerBooter, BootComponent} from '../../index';
+import {resolve} from 'path';
+
+describe('controller booter acceptance tests', () => {
+  // tslint:disable-next-line:no-any
+  let app: any;
+  const SANDBOX_PATH = resolve(__dirname, '../sandbox');
+  const sandbox = new TestSandbox(SANDBOX_PATH);
+  const bootConfig = {
+    projectRoot: SANDBOX_PATH,
+  };
+
+  beforeEach(resetSandbox);
+  beforeEach(getApp);
+
+  afterEach(stopApp);
+
+  it('binds controllers using ControllerDefaults and REST endpoints work', async () => {
+    await app.booter(ControllerBooter);
+    await app.boot(bootConfig);
+    await app.start();
+
+    const server: RestServer = await app.getServer(RestServer);
+    const client: Client = createClientForHandler(server.handleHttp);
+
+    // Default Controllers = /controllers with .controller.js ending (nested = true);
+    await client.get('/').expect(200, 'HelloController.hello()');
+    await client.get('/one').expect(200, 'ControllerOne.one()');
+    await client.get('/two').expect(200, 'ControllerOne.two()');
+  });
+
+  async function getApp() {
+    await sandbox.copyFile(resolve(__dirname, '../fixtures/application.js'));
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/application.js.map'),
```

Comment:
Is this file needed for the test? Isn't the `map` file going to point to a wrong place (assuming it uses relative path to the original `.ts` file)?

Same comment applies to other `.map` files below.

Reply:
The lack of moving the `.map` files was producing a warning message that was being logged when the tests are run. I was trying to keep the console output clean ... but the tests still pass and the `.map` files aren't really needed. I'll remove them.

Reply:
Oh, are the warnings saying that `.js` files created by `tsc` are referencing external `.map` files that do not exist? That would make sense! Can we copy the `.ts` files too?

> I was trying to keep the console output clean 

Please keep keeping the console output clean :)

Reply:
This is the message that gets logged to the console (test still passes). 
```
Transformation error for /Users/taranveer/loopback/loopback-next/packages/boot/dist6/test/sandbox/application.js ; return original code
An error occurred while trying to read the map file at /Users/taranveer/loopback/loopback-next/packages/boot/dist6/test/sandbox/application.js.map
Error: ENOENT: no such file or directory, open '/Users/taranveer/loopback/loopback-next/packages/boot/dist6/test/sandbox/application.js.map'
```

Reply:
I am proposing to create a helper method that will copy both `.js` and `.js.map` files, so that we don't have to repeat ourselves so much. 

Example usage:

```ts
await sandbox.copySourceCodeFile(
  resolve(__dirname, '../fixtures/hello.controller.js'),
  'controllers/nested/'
  // ^-- target filename is the same as the original name
  // we can use the trailing `/` (or missing file extension?) to trigger this logic
);
```

Reply:
Out of scope of MVP. I'll add a comment to the WIP PR to update the tests in Booter post-MVP when that PR is picked up on / continued.

Reply:
I disagree, see my comment https://github.com/strongloop/loopback-next/pull/951#issuecomment-362522957.

However, I don't want to block progress on this. As long as you add the comment to these tests pointing to your pull request in progress, I can live with the current code proposed here.